ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql

```diff
@@ -0,0 +1,38 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+
+
+CREATE TABLE participant_users (
+    internal_id         SERIAL          PRIMARY KEY,
+    user_id             VARCHAR(256)            NOT NULL UNIQUE,
+    primary_party       VARCHAR(512),
+    -- The default is  microseconds since Unix epoch
+    created_at          BIGINT          NOT NULL DEFAULT (1000 * 1000 * EXTRACT(epoch FROM CURRENT_TIMESTAMP))::bigint
+);
+COMMENT ON COLUMN participant_users.created_at IS
+    'Up to and including Postgres v13 EXTRACT function returns double type and thus created_at is susceptible to microseconds rounding errors from around year 2250. In contrast, Postgres v14 EXTRACT returns numeric type which effectively has arbitrary precision for this use case';
+
+
+CREATE TABLE participant_user_rights (
+    user_internal_id    INTEGER         NOT NULL REFERENCES participant_users (internal_id) ON DELETE CASCADE,
+    user_right          INTEGER         NOT NULL,
+    for_party           VARCHAR(512),
+    -- The default is microseconds since Unix epoch
+    granted_at          BIGINT          NOT NULL DEFAULT (1000 * 1000 * EXTRACT(epoch FROM CURRENT_TIMESTAMP))::bigint,
+    UNIQUE (user_internal_id, user_right, for_party)
+);
+-- Creating additional partial index to ensure at most one NULL for_porty for each unique (user_internal_id, user_right) pair
+CREATE UNIQUE INDEX participant_user_rights_user_internal_id_user_right_idx
+    ON participant_user_rights (user_internal_id, user_right)
+    WHERE for_party IS NULL;
```

Comment:
Good catch @pbatko-da . Given the structure of rights we have, consider the option of requiring `for_party NOT NULL` and using `""` for `participant_admin` rights. Essentially using Protobuf's defaulting rules to not deal with NULL's at all. This would remove the need for partial indices, and thus reduce both development and storage overhead.

I've taken the inspiration from: https://dba.stackexchange.com/questions/299098/why-doesnt-my-unique-constraint-trigger/299107#299107

Reply:
Thanks, will try to use empty strings

Reply:
Done